UefiCpuPkg/PiSmmCpuDxeSmm/NonMmramMapStandaloneMm.c

```diff
@@ -138,11 +141,22 @@ BuildMemoryMapFromResDescHobs (
   Hob.Raw = GetFirstHob (EFI_HOB_TYPE_RESOURCE_DESCRIPTOR);
   while (Hob.Raw != NULL) {
     if ((Hob.ResourceDescriptor->ResourceAttribute & MM_RESOURCE_ATTRIBUTE_LOGGING) == 0) {
-      //
-      // Resource HOBs describe all accessible non-smram regions.
-      // Logging attribute range is treated as not present. Not-present ranges are not included in this memory map.
-      //
-      Count++;
+      ResourceHobEnd = Hob.ResourceDescriptor->PhysicalStart + Hob.ResourceDescriptor->ResourceLength;
+
+      if (ResourceHobEnd <= MaxPhysicalAddress) {

```

Comment:
Just need assert:
ASSERT (Hob.ResourceDescriptor->PhysicalStart < MaxPhysicalAddress && ResourceHobEnd < MaxPhysicalAddress)

Reply:
what if the unexpected behavior happens in release build?

Reply:
for any issue check, we will use the debug version. Here, it's just help us quick find the problem. debug log will also not output in release version.